driftbase/api/flexmatch.py

```diff
@@ -0,0 +1,117 @@
+"""
+    Orchestration of GameLift/FlexMatch matchmaking
+"""
+
+from flask_smorest import Blueprint, abort
+from drift.core.extensions.urlregistry import Endpoints
+from drift.core.extensions.jwt import requires_roles
+from marshmallow import Schema, fields
+from flask.views import MethodView
+from flask import url_for, request
+from six.moves import http_client
+from drift.core.extensions.jwt import current_user
+from driftbase import flexmatch
+import logging
+
+
+
+bp = Blueprint("flexmatch", "flexmatch", url_prefix="/flexmatch", description="Orchestration of GameLift/FlexMatch matchmaking.")
+endpoints = Endpoints()
+log = logging.getLogger(__name__)
+
+def drift_init_extension(app, api, **kwargs):
+    api.register_blueprint(bp)
+    endpoints.init_app(app)
+
+
+class FlexMatchPlayerAPIPatchArgs(Schema):
+    latency_ms = fields.Float(required=True, description="Latency between client and the region he's measuring against.")
+    region = fields.String(required=True, description="Which region the latency was measured against.")
+
+class FlexMatchPlayerAPIPostArgs(Schema):
+    matchmaker = fields.String(required=True, description="Which matchmaker (configuration name) to issue the ticket for. ")
+
+class FlexMatchPlayerAPIPutArgs(Schema):
+    match_id = fields.String(required=True, description="The id of the match being accepted/rejected")
+    acceptance = fields.Boolean(required=True, description="True if match_id is accepted, False otherwise")
+
+
+@bp.route("/")
+class FlexMatchPlayerAPI(MethodView):
+
+    @bp.arguments(FlexMatchPlayerAPIPatchArgs)
+    def patch(self, args):
+        """
+        Add a freshly measured latency value to the player tally.
+        Returns a region->avg_latency mapping.
+        """
+        player_id = current_user["player_id"]
+        latency = args.get("latency_ms")
+        region = args.get("region")
+        if None in (latency, region) or region not in flexmatch.VALID_REGIONS or not isinstance(latency, (int, float)):
```

Comment:
the client has config entries specifying which servers to ping for which region, and I'm thinking when we go do multi-region, it should query the backend for these region->pingserver mapping.  Until we get to that I didn't see the point in putting it in the config.

Reply:
Well, if we want to experiment now, we'd need to re-deploy the backend instead of just updating the config. Since there's already a config resource, it should be a quick change, even if the API is not there for the client just yet